### ISSUE TITLE
Timeout error returns non-valid Javascript object

### ISSUE THREAD

**Issue by jackspaniel** [forwardemail/superagent#744]
Maybe I am missing something here. But here is what I see in the err argument for a call that times out:

```
call.end(function(err, response) {
    console.log(err); // { [Error: timeout of 1ms exceeded] timeout: 1 }  
    console.log(JSON.stringify(err)); // {"timeout":1}
    console.log(typeof err); // object
```

It seems like the error message is somehow getting tacked onto the err object w/o a proper key?

**Comment by jackspaniel** [comment]
Ok well a little further research shows all err objects looking similar: 

    call.end(function(err, response) {

        console.log(err); // { [Error: getaddrinfo ENOTFOUND xxxhttp] code: 'ENOTFOUND', errno: 'ENOTFOUND',  syscall: 'getaddrinfo',  hostname: 'xxxhttp' }

        console.log(JSON.stringify(err)); // {"code":"ENOTFOUND","errno":"ENOTFOUND","syscall":"getaddrinfo","hostname":"xxxhttp"}

        console.log(typeof err); // object

Basically I am trying to isolate timeout errors. Can I trust that if the timeout property exists in the error object, that means timeout error and nothing else? I don't really trust doing err.toString or something weird to see the error message. Ideally it would be nice to have an error code like code; "TIMEOUT".

**Comment by jackspaniel** [comment]
Ok well now I'm guessing you just overloaded the toString() method. Although I don't see that in the source code anywhere.

**Comment by codeviking** [comment]
I ran into this as well, but if you check this out...

https://github.com/visionmedia/superagent/blob/master/lib/client.js#L881

...depending on the existence of `err.timeout` appears to be reliable, I'm going with that for now.

**Comment by titanism** [comment]
This is normal Error serialization: JSON.stringify omits non-enumerable message. Detect timeouts with err && err.code === 'ECONNABORTED' and, if needed, err.timeout.

v10.4.0 is available on npm: https://www.npmjs.com/package/superagent/v/10.4.0
GitHub release: https://github.com/forwardemail/superagent/releases/tag/v10.4.0

Consider supporting our efforts to maintain this project by using Forward Email at https://forwardemail.net.  We run the only 100% open source and 3rd party audited email service in the world.

**Comment by titanism** [comment]
Closing this for now. If this still affects v10.4.0, please open a new issue with a current reproduction.

v10.4.0 is available on npm: https://www.npmjs.com/package/superagent/v/10.4.0
GitHub release: https://github.com/forwardemail/superagent/releases/tag/v10.4.0

Consider supporting our efforts to maintain this project by using Forward Email at https://forwardemail.net.  We run the only 100% open source and 3rd party audited email service in the world.
